backend/src/folder/folder.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@nestjs/common';
+import { CreateFolderInput } from './dto/create-folder.input';
+import { UpdateFolderInput } from './dto/update-folder.input';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { Folder } from './entities/folder.entity';
+
+@Injectable()
+export class FolderService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async getFolders(userId: string): Promise<Folder[]> {
+    return this.prisma.folder.findMany({
+      where: { userId },
+    });
+  }
+
+  async getFolder(id: string, userId: string): Promise<Folder> {
+    const folder = await this.prisma.folder.findUnique({
+      where: { id, userId },
+      include: { journals: true } as any,
+    });
+
+    if (!folder) {
+      throw new Error('Folder not found or access denied');
```

Comment:
Since your `getFolder` mutation is nullable — I would return null here if folder is not found

If you want to throw an Error (which is totally fine) — change your query type to not be nullable

Reply:
I return null here if folder is not found instead